tensorflow_zero_out/cc/kernels/lsh_matmul.cc

```diff
@@ -0,0 +1,221 @@
+    return rawResults;
+  }
+
+  int getCandidates(int **actives) {
+    for(int i = 0; i < _L; i++) {
+      if(actives[i] == NULL) {
+        continue;
+      } 
+      else {
+        for(int j = 0; j < BUCKETSIZE; j++) {
+          int tempID = actives[i][j] - 1;
+          if(tempID >= 0) 
+            counts[tempID] += 1;
+          else
+            break;
+        }
+      }
+    }
+
+    if(counts.size()<1500){
+      srand(time(NULL));
+      int start = rand() % _noOfNodes;
+      for(int i = start; i < _noOfNodes; i++) {
+        if(counts.size() >= 1000) 
+          break;
+        if(counts.count(_randNode[i]) == 0)
+          counts[_randNode[i]] = 0;
+      }
+
+
+      if(counts.size() < 1000) {
+        for(int i = 0; i < _noOfNodes; i++) {
+          if(counts.size() >= 1000)
+            break;
+          if(counts.count(_randNode[i]) == 0)
+            counts[_randNode[i]] = 0;
+        }
+      }
+    }
+
+    len = counts.size();
+    return len;
+  }
+};
+
+void Compute(OpKernelContext* ctx) override {
+    // Note:
+    // need values(x), weights, indices, inputID,  activenodesPerLayer, 
+    // activeValues & length as input parameters or attributes
+    const Tensor& a = ctx->input(0);
+    Tensor an = ctx->input(1);
+    Tensor av = ctx->input(2);
+    Tensor l = ctx->input(3);
+    Tensor i = ctx->input(4);
+    Tensor w = ctx->input(5);
+    Tensor in = ctx->input(6);
+
+    // Note: Not sure if we need transpose. 
+    // TODO: check back later
+    OP_REQUIRES(ctx, TensorShapeUtils::IsMatrix(a.shape()),
+                errors::InvalidArgument("In[0] is not a matrix"));
+    OP_REQUIRES(ctx, TensorShapeUtils::IsMatrix(w.shape()),
+                errors::InvalidArgument("In[1] is not a matrix"));
+    Eigen::array<Eigen::IndexPair<Eigen::DenseIndex>, 1> dim_pair;
+    dim_pair[0].first = transpose_a_ ? 0 : 1;
+    dim_pair[0].second = transpose_w_ ? 1 : 0;
+
+    OP_REQUIRES(
+        ctx, a.dim_size(dim_pair[0].first) == w.dim_size(dim_pair[0].second),
+        errors::InvalidArgument(
+            "Matrix size-incompatible: In[0]: ", a.shape().DebugString(),
+            ", In[1]: ", w.shape().DebugString()));
+    int a_dim_remaining = 1 - dim_pair[0].first;
+    int w_dim_remaining = 1 - dim_pair[0].second;
+    TensorShape out_shape(
+        {a.dim_size(a_dim_remaining), w.dim_size(w_dim_remaining)});
+    Tensor* out = nullptr;
+    OP_REQUIRES_OK(ctx, ctx->allocate_output(0, out_shape, &out));
+    if (out->NumElements() == 0) {
+      // If a has shape [0, x] or b has shape [x, 0], the output shape
+      // is a 0-element matrix, so there is nothing to do.
+      return;
+    }
+
+    auto  weights = (w.template flat<T>().data());
+    auto  indices = (i.template flat<T>().data());
+    auto  inputID = (in.template flat<T>().data());
+    auto  length = (l.template flat<T>().data());
+    auto  activeNodesPerLayer = (an.template flat<T>().data());
+    auto  activeValues = (av.template flat<T>().data());
```

Comment:
I think better to follow tensor flow's coding conventions here and call it input_id than inputID. Same for other variable names.